src/controllers/expenses.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const expenseService = require('../services/expenses');
+const userService = require('../services/users');
+
+const getAll = async(req, res) => {
+  const { userId, categories, from, to } = req.query;
+  let expenses = await expenseService.getAll();
+
+  if (userId) {
+    expenses = expenses.filter((expense) => expense.userId === +userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter((expense) =>
+      categories.includes(expense.category),
+    );
+  }
+
+  if (from && to) {
+    expenses = expenses.filter((expense) => {
+      const normalizedExpenseDate = new Date(expense.spentAt);
+      const normalizedDateFrom = new Date(from);
+      const normalizedDateTo = new Date(to);
+
+      return (
+        normalizedExpenseDate >= normalizedDateFrom
+        && normalizedExpenseDate <= normalizedDateTo
+      );
+    });
+  }
+
```

Comment:
consider moving all the complex logic to service file as controller is just a middleware between service and route so it shouldn't have this logic